### ISSUE TITLE
Wrong Type parameters.

### ISSUE THREAD

**Issue by jDramaix** [angular/clutz#269]
The type parameters seems not to be converted correctly. Let’s take the Array example and compare the definition of the type in extern and its conversion in typescript:

Es3.js (for the sake of the example, we just keep JsDoc annotations that are relevant for the issue)

```
/**
 * @constructor
 * @implements {IArrayLike<T>}
 * @template T
 */
function Array(var_args) {}

/**
 * @record
 * @extends {IObject<number, VALUE2>}
 * @template VALUE2
 */
function IArrayLike() {}

/**
 * @interface
 * @template KEY1, VALUE1
 */
function IObject() {}
```

generated .d.ts code:

```
class Array<IObject#VALUE, T> extends Array_Instance<IObject#VALUE, T> {}

class Array_Instance<IObject#VALUE, T> implements ArrayLike<T> {}

interface IArrayLike<VALUE2> extends IObject<number, VALUE2> {}

interface IObject<IObject#KEY1, IObject#VALUE> {}
```

Note the ```#``` in the type parameter identifier that is not valid in typescript.

**Comment by rkirov** [comment]
This exposes some internal Closure magic, that is not visible from the actual definitions in es3.js. I don't understand closure internals well enough to reason why they do this, but one simple path forward is to just strip everything after `#`. I will prep a cl for that.

**Comment by jDramaix** [comment]
After some investigations, this is a problem in the JsCompiler itself and are very specific to `Array` and `Object` types. I will open a bug to JsCompiler product and I can propose a workaround in clutz to avoid this problem.

**Comment by jDramaix** [comment]
I have implemented a workaround: https://github.com/jDramaix/clutz/pull/1/files (it seems I cannot stack pull request on the upsteam repository with github?)

**Comment by rkirov** [comment]
Great, I don't see a reason why your can't create a PR against angular/clutz (not jDramaix/clutz). Worst to worst we can do the review on https://github.com/jDramaix/clutz/pull/1/files and I will just cherry pick it in. I left a comment there.

Also, for your information, a few months ago, Martin and I brainstormed on emitting all closure externs, similar to what you are trying to do. Eventually, we went with the "don't emit and reuse lib.d.ts" approach, but I did prototype that attempt in this branch - https://github.com/angular/clutz/commit/2047be88724ddf0a86490ebf974fc388178d7ef6
There I ran into all the same issues you are seeing now, so it might be of some use to you.
